source/scpd/Cpp.d

```diff
@@ -44,6 +44,7 @@ extern(C++, (StdNamespace)) {
             private alias TPtr = T*;
 
         ~this() @safe {}
+        this (const ref shared_ptr!T) @safe nothrow;
```

Comment:
Weirdly, this links fine on Linux with both GCC and Clang runtimes but fails on MacOS

Reply:
I'm pretty sure it gets inlined on Mac OSX. I'm surprised it does not on Linux, that sounds absurdly inefficient. 
This is the definition on my machine. Note the `inline`:
```C++
template<class _Tp>
template<class _Yp>
inline
shared_ptr<_Tp>::shared_ptr(const shared_ptr<_Yp>& __r,
                            typename enable_if<__compatible_with<_Yp, element_type>::value, __nat>::type)
         _NOEXCEPT
    : __ptr_(__r.__ptr_),
      __cntrl_(__r.__cntrl_)
{
    if (__cntrl_)
        __cntrl_->__add_shared();
}
```